recipes/libid3tag/all/conanfile.py

```diff
@@ -1,124 +1,116 @@
 import os
-import shutil
 
-required_conan_version = ">=1.33.0"
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMakeToolchain, CMakeDeps, CMake
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import chdir, copy, get, rm, replace_in_file
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc
+
+required_conan_version = ">=1.53.0"
 
 
 class LibId3TagConan(ConanFile):
     name = "libid3tag"
     description = "ID3 tag manipulation library."
-    topics = ("conan", "mad", "id3", "MPEG", "audio", "decoder")
+    license = "GPL-2.0-or-later"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.underbit.com/products/mad/"
-    license = "GPL-2.0-or-later"
-    settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": False, "fPIC": True}
-    generator = "pkg_config", "visual_studio"
-
-    _autotools = None
+    topics = ("mad", "id3", "MPEG", "audio", "decoder")
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", self.recipe_folder, os.path.join(self.export_sources_folder, "src"))
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
-    def requirements(self):
-        self.requires("zlib/1.2.11")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
-    @property
-    def _is_msvc(self):
-        return self.settings.compiler == "Visual Studio" or (
-            self.settings.compiler == "clang" and self.settings.os == "Windows"
-        )
+    def requirements(self):
+        self.requires("zlib/[>=1.2.11 <2]")
 
     def validate(self):
-        if self._is_msvc and self.options.shared:
-            raise ConanInvalidConfiguration("libid3tag does not support shared library for MSVC")
-
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+        if self.settings.arch == "armv8" and self.options.shared:
+            raise ConanInvalidConfiguration("shared library build is not supported for armv8")
```

Comment:
```suggestion
```
I suggest trying it again. I can build it in Mac M1 locally, but probably you had problems when cross-building from Mac Intel to Mac M1. However, we added Mac M1 to the build matrix, so it should work now.